opts/opts.go

```diff
@@ -302,6 +302,26 @@ func ValidateSysctl(val string) (string, error) {
 	return "", fmt.Errorf("sysctl '%s' is not whitelisted", val)
 }
 
+// ValidateProxyEnv checks proxy environment variables for validity
+func ValidateProxyEnv(val string) (string, error) {
+	permitted := []string{"HTTP_PROXY", "HTTPS_PROXY", "NO_PROXY", "FTP_PROXY"}
```

Comment:
I think this is the current list we support for `--build-arg`; if we decide to add more, we should keep those in sync (should be a separate PR / discussion) ; maybe we can use the same list (from code) to guarantee they stay in sync?

Note that these are (afaik) not standardized and there's variations of these (uppercase/lowercase); some software uses lowercase, other uppercase 😞